### ISSUE TITLE
k8s v1.15-beta1: kubelet: unknown flag: --allow-privileged

### ISSUE THREAD

**Issue by tstromberg** [kubernetes/minikube#4371]
If `minikube start --kubernetes-version=v1.15.0-beta.1` is used, a failure occurs:

```

Unfortunately, an error has occurred:
	timed out waiting for the condition

This error is likely caused by:
	- The kubelet is not running
	- The kubelet is unhealthy due to a misconfiguration of the node in some way (required cgroups disabled)

If you are on a systemd-powered system, you can try to troubleshoot the error with the following commands:
	- 'systemctl status kubelet'
	- 'journalctl -xeu kubelet'

Additionally, a control plane component may have crashed or exited when started by the container runtime.
To troubleshoot, list all containers using your preferred container runtimes CLI, e.g. docker.
Here is one example how you may list all Kubernetes containers running in docker:
	- 'docker ps -a | grep kube | grep -v pause'
	Once you have found the failing container, you can inspect its logs with:
	- 'docker logs CONTAINERID'

: Process exited with status 1

😿  Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
👉  https://github.com/kubernetes/minikube/issues/new
❌  Problems detected in "kubelet":
    May 29 18:23:32 k8s115 kubelet[2944]: F0529 18:23:32.984077    2944 server.go:156] unknown flag: --allow-privileged
```

**Comment by tstromberg** [comment]
Deprecation notice for this flag: https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.15.md#action-required

If anyone wants to fix this, here is where to get started:

https://github.com/kubernetes/minikube/blob/6bc6c3fe876f55534360bc1a8aa30e1722f388ce/pkg/minikube/bootstrapper/kubeadm/versions.go#L245

If Kubernetes v1.15.0-beta1 is working properly, the PR should also update NewestKubernetesVersion so that this release is tested in our integration tests:

https://github.com/kubernetes/minikube/blob/b64f62ccfb5ec34897540b1721fa3c89b5c8c5b5/pkg/minikube/constants/constants.go#L168

**Comment by blueelvis** [comment]
Am I correct in understanding that I would have to remove the above line 245 which you have mentioned and add code similar to this https://github.com/kubernetes/minikube/blob/16af8b403c3c97af6f018cc4a24b125b53693af8/pkg/minikube/bootstrapper/kubeadm/versions.go#L257  so that this is set only for versions which support it? (Basically setting the `LessThanOrEqual`

**Comment by fenglixa** [comment]
I would like to provide PR for this ticket. Thanks @blueelvis @tstromberg

**Comment by the-redback** [comment]
Kubernetes-1.15.0 unable to run on minikube 1.1.1 as this error still exists on last release. 
Is there any early plan of release? @tstromberg

**Comment by fenglixa** [comment]
@the-redback Let me check.

**Comment by fenglixa** [comment]
With leatest code, minikube could start kubenetes-1.15.0 successfully, as below my test result:
![image](https://user-images.githubusercontent.com/12505594/59986235-72e6bc00-9668-11e9-8916-4b5b349c0a63.png)

@the-redback  I am worried the fixes was not in the build you are using, wish you could help double check. Thanks.

**Comment by tstromberg** [comment]
@the-redback - minikube v1.2.0 with v1.15.0 support will be published this week.

**Comment by the-redback** [comment]
Thanks @tstromberg .

**Comment by the-redback** [comment]
@fenglixa , It's strange see the log I am getting below:

```
minikube start --kubernetes-version=v1.15.0                                        
😄  minikube v1.1.1 on linux (amd64)
🔥  Creating virtualbox VM (CPUs=2, Memory=4096MB, Disk=20000MB) ...
🐳  Configuring environment for Kubernetes v1.15.0 on Docker 18.09.6
🚜  Pulling images ...
🚀  Launching Kubernetes ... 

💣  Error starting cluster: cmd failed: sudo /usr/bin/kubeadm init --config /var/lib/kubeadm.yaml  --ignore-preflight-errors=DirAvailable--etc-kubernetes-manifests,DirAvailable--data-minikube,FileAvailable--etc-kubernetes-manifests-kube-scheduler.yaml,FileAvailable--etc-kubernetes-manifests-kube-apiserver.yaml,FileAvailable--etc-kubernetes-manifests-kube-controller-manager.yaml,FileAvailable--etc-kubernetes-manifests-etcd.yaml,Port-10250,Swap
[init] Using Kubernetes version: v1.15.0
[preflight] Running pre-flight checks
	[WARNING Service-Docker]: docker service is not enabled, please run 'systemctl enable docker.service'
	[WARNING IsDockerSystemdCheck]: detected "cgroupfs" as the Docker cgroup driver. The recommended driver is "systemd". Please follow the guide at https://kubernetes.io/docs/setup/cri/
	[WARNING Service-Kubelet]: kubelet service is not enabled, please run 'systemctl enable kubelet.service'
[preflight] Pulling images required for setting up a Kubernetes cluster
[preflight] This might take a minute or two, depending on the speed of your internet connection
[preflight] You can also perform this action in beforehand using 'kubeadm config images pull'
[kubelet-start] Writing kubelet environment file with flags to file "/var/lib/kubelet/kubeadm-flags.env"
[kubelet-start] Writing kubelet configuration to file "/var/lib/kubelet/config.yaml"
[kubelet-start] Activating the kubelet service
[certs] Using certificateDir folder "/var/lib/minikube/certs/"
[certs] Generating "etcd/ca" certificate and key
[certs] Generating "etcd/server" certificate and key
[certs] etcd/server serving cert is signed for DNS names [minikube localhost] and IPs [192.168.99.132 127.0.0.1 ::1]
[certs] Generating "etcd/peer" certificate and key
[certs] etcd/peer serving cert is signed for DNS names [minikube localhost] and IPs [192.168.99.132 127.0.0.1 ::1]
[certs] Generating "etcd/healthcheck-client" certificate and key
[certs] Generating "apiserver-etcd-client" certificate and key
[certs] Generating "front-proxy-ca" certificate and key
[certs] Generating "front-proxy-client" certificate and key
[certs] Using existing ca certificate authority
[certs] Using existing apiserver certificate and key on disk
[certs] Generating "apiserver-kubelet-client" certificate and key
[certs] Generating "sa" key and public key
[kubeconfig] Using kubeconfig folder "/etc/kubernetes"
[kubeconfig] Writing "admin.conf" kubeconfig file
[kubeconfig] Writing "kubelet.conf" kubeconfig file
[kubeconfig] Writing "controller-manager.conf" kubeconfig file
[kubeconfig] Writing "scheduler.conf" kubeconfig file
[control-plane] Using manifest folder "/etc/kubernetes/manifests"
[control-plane] Creating static Pod manifest for "kube-apiserver"
[control-plane] Creating static Pod manifest for "kube-controller-manager"
[control-plane] Creating static Pod manifest for "kube-scheduler"
[etcd] Creating static Pod manifest for local etcd in "/etc/kubernetes/manifests"
[wait-control-plane] Waiting for the kubelet to boot up the control plane as static Pods from directory "/etc/kubernetes/manifests". This can take up to 4m0s
[kubelet-check] Initial timeout of 40s passed.
[kubelet-check] It seems like the kubelet isn't running or healthy.
[kubelet-check] The HTTP call equal to 'curl -sSL http://localhost:10248/healthz' failed with error: Get http://localhost:10248/healthz: dial tcp 127.0.0.1:10248: connect: connection refused.
[kubelet-check] It seems like the kubelet isn't running or healthy.
[kubelet-check] The HTTP call equal to 'curl -sSL http://localhost:10248/healthz' failed with error: Get http://localhost:10248/healthz: dial tcp 127.0.0.1:10248: connect: connection refused.
[kubelet-check] It seems like the kubelet isn't running or healthy.
[kubelet-check] The HTTP call equal to 'curl -sSL http://localhost:10248/healthz' failed with error: Get http://localhost:10248/healthz: dial tcp 127.0.0.1:10248: connect: connection refused.
[kubelet-check] It seems like the kubelet isn't running or healthy.
[kubelet-check] The HTTP call equal to 'curl -sSL http://localhost:10248/healthz' failed with error: Get http://localhost:10248/healthz: dial tcp 127.0.0.1:10248: connect: connection refused.
error execution phase wait-control-plane: couldn't initialize a Kubernetes cluster
[kubelet-check] It seems like the kubelet isn't running or healthy.
[kubelet-check] The HTTP call equal to 'curl -sSL http://localhost:10248/healthz' failed with error: Get http://localhost:10248/healthz: dial tcp 127.0.0.1:10248: connect: connection refused.

Unfortunately, an error has occurred:
	timed out waiting for the condition

This error is likely caused by:
	- The kubelet is not running
	- The kubelet is unhealthy due to a misconfiguration of the node in some way (required cgroups disabled)

If you are on a systemd-powered system, you can try to troubleshoot the error with the following commands:
	- 'systemctl status kubelet'
	- 'journalctl -xeu kubelet'

Additionally, a control plane component may have crashed or exited when started by the container runtime.
To troubleshoot, list all containers using your preferred container runtimes CLI, e.g. docker.
Here is one example how you may list all Kubernetes containers running in docker:
	- 'docker ps -a | grep kube | grep -v pause'
	Once you have found the failing container, you can inspect its logs with:
	- 'docker logs CONTAINERID'

: Process exited with status 1

😿  Sorry that minikube crashed. If this was unexpected, we would love to hear from you:
👉  https://github.com/kubernetes/minikube/issues/new
❌  Problems detected in "kubelet":
    Jun 24 03:42:49 minikube kubelet[3337]: F0624 03:42:49.281947    3337 server.go:156] unknown flag: --allow-privileged
```

**Comment by the-redback** [comment]
@fenglixa, I see. you are using custom build instead of last stable release, right? That's what I was reporting in the first place. Then I got confused seeing your log. :)

anyway, end of misunderstanding. I am okay to wait one more week for next release. :D

**Comment by fenglixa** [comment]
Yes, I am using source code build, so from @tstromberg, this fix will be included in v1.2.0.

**Comment by tstromberg** [comment]
minikube v1.2.0 is released, and defaults to Kubernetes v1.1.15. Thank you for your patience!
